packages/parameters/src/base/transformValue.ts

```diff
@@ -0,0 +1,64 @@
+import type { JSONValue } from '@aws-lambda-powertools/commons';
+import { isString } from '@aws-lambda-powertools/commons';
+import { fromBase64 } from '@aws-sdk/util-base64-node';
+import { TRANSFORM_METHOD_BINARY, TRANSFORM_METHOD_JSON } from '../constants';
+import { TransformParameterError } from '../errors';
+import type { TransformOptions } from '../types/BaseProvider';
+
+/**
+ * Utility function to transform a value.
+ *
+ * It supports JSON and binary transformations, as well as an 'auto' mode that will try to transform the value based on the key.
```

Comment:
I'd add signals on the logic of this transformation when parameter ends with `.json`, `.binary`, so the user can understand which inputs will be transformed. Not too details of course to expose and repeat entire implementation.

Reply:
Updated in the last commit, explaining the flow. Good suggestion